internal/kvrpc/batch.go

```diff
@@ -34,25 +35,30 @@ type BatchResult struct {
 
 // AppendBatches divides the mutation to be requested into Batches so that the size of each batch is
 // approximately the same as the given limit.
-func AppendBatches(batches []Batch, regionID locate.RegionVerID, groupKeys [][]byte, keyToValue map[string][]byte, limit int) []Batch {
+func AppendBatches(batches []Batch, regionID locate.RegionVerID, groupKeys [][]byte, keyToValue map[string][]byte, keyToTtl map[string]uint64, limit int) []Batch {
 	var start, size int
 	var keys, values [][]byte
+	var ttls []uint64
 	for start = 0; start < len(groupKeys); start++ {
 		if size >= limit {
-			batches = append(batches, Batch{RegionID: regionID, Keys: keys, Values: values})
+			batches = append(batches, Batch{RegionID: regionID, Keys: keys, Values: values, Ttls: ttls})
 			keys = make([][]byte, 0)
 			values = make([][]byte, 0)
+			ttls = make([]uint64, 0)
 			size = 0
 		}
 		key := groupKeys[start]
 		value := keyToValue[string(key)]
+		ttl := keyToTtl[string(key)]
 		keys = append(keys, key)
 		values = append(values, value)
+		ttls = append(ttls, ttl)
 		size += len(key)
 		size += len(value)
+		size += 8
```

Comment:
I think we can ignore the size of TTL as protobuf uses varint to encode integers. Typically it won't cost 8 bytes.